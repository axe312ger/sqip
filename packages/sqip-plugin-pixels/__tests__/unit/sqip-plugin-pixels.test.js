import sqipPluginPixels from '../../src/sqip-plugin-pixels'
import { resolve } from 'path'

const FILE_DEMO_BEACH = resolve(__dirname, '../../../..', 'demo', 'beach.jpg')

describe('sqip-plugin-pixels', () => {
  it('turns raster image into pixelated svg', async () => {
    const plugin = new sqipPluginPixels({ input: FILE_DEMO_BEACH })
    const result = await plugin.apply()
    expect(result).toMatchSnapshot()
  })
})
