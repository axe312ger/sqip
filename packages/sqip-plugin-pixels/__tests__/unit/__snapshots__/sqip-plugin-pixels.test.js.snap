// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqip-plugin-pixels turns raster image into pixelated svg 1`] = `"<svg xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xmlns:svgjs=\\"http://svgjs.com/svgjs\\" width=\\"800\\" height=\\"500\\"><rect width=\\"100\\" height=\\"100\\" fill=\\"#5489b0\\" x=\\"0\\" y=\\"0\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#538db5\\" x=\\"100\\" y=\\"0\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#7493a6\\" x=\\"200\\" y=\\"0\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#a89a8d\\" x=\\"300\\" y=\\"0\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#b49a84\\" x=\\"400\\" y=\\"0\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#9f9184\\" x=\\"500\\" y=\\"0\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#768f9c\\" x=\\"600\\" y=\\"0\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#558db1\\" x=\\"700\\" y=\\"0\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#729cb8\\" x=\\"0\\" y=\\"100\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#739fbc\\" x=\\"100\\" y=\\"100\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#8da8b1\\" x=\\"200\\" y=\\"100\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#acaea1\\" x=\\"300\\" y=\\"100\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#b3a188\\" x=\\"400\\" y=\\"100\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#b3a38a\\" x=\\"500\\" y=\\"100\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#a5a698\\" x=\\"600\\" y=\\"100\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#7d9fb2\\" x=\\"700\\" y=\\"100\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#5f8fa9\\" x=\\"0\\" y=\\"200\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#6192ab\\" x=\\"100\\" y=\\"200\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#5c93ae\\" x=\\"200\\" y=\\"200\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#5895b4\\" x=\\"300\\" y=\\"200\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#5891ac\\" x=\\"400\\" y=\\"200\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#5c8ba6\\" x=\\"500\\" y=\\"200\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#5e839a\\" x=\\"600\\" y=\\"200\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#558eaa\\" x=\\"700\\" y=\\"200\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#457b91\\" x=\\"0\\" y=\\"300\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#467e93\\" x=\\"100\\" y=\\"300\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#488093\\" x=\\"200\\" y=\\"300\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#4a8193\\" x=\\"300\\" y=\\"300\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#528694\\" x=\\"400\\" y=\\"300\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#6a5248\\" x=\\"500\\" y=\\"300\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#605b56\\" x=\\"600\\" y=\\"300\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#4b8a9e\\" x=\\"700\\" y=\\"300\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#a1aba4\\" x=\\"0\\" y=\\"400\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#a7b0a7\\" x=\\"100\\" y=\\"400\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#acb5aa\\" x=\\"200\\" y=\\"400\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#b5bdb0\\" x=\\"300\\" y=\\"400\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#989587\\" x=\\"400\\" y=\\"400\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#534340\\" x=\\"500\\" y=\\"400\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#525a5d\\" x=\\"600\\" y=\\"400\\"></rect><rect width=\\"100\\" height=\\"100\\" fill=\\"#656f6f\\" x=\\"700\\" y=\\"400\\"></rect></svg>"`;
